### ISSUE TITLE
Proposed Changes on Maybe

### ISSUE THREAD

**Issue by kaifox** [ossgang/ossgang-commons#18]
Here just some proposals, as I started using maybe:

In parallel to the method `onException(Consumer<Exception> c);`, we could have a method

```
onValue(Consumer<T> c);
```
(If we would stick closer to optional, then we would even call them 
```
Maybe<T> ifException(Consumer<Throwable> c);
Maybe<T> ifValue(Consumer<T> c);
```

In the end it should be equvalent to chain either
```
m.ifException(e->handleException(e)).ifValue(v->useValue(v));
```
or
```
m.ifValue(v->useValue(v)).ifException(e->handleException(e));
```

Further, the value() method returns T, while the exception() method returns an optional Exception. More consistent I would find:

```
T value(); // exists
Optional<T> optionalValue(); // exists
Throwable exception();
Optional<Throwable> optionalException();
```

**Comment by michi42** [comment]
Thanks for the proposals! As discussed in Slack - go ahead ;-)

**Comment by andreacalia** [comment]
Looks good the code Kajetan!
Do you think we can close this issue?

**Comment by michi42** [comment]
Released as 0.5.0.
